druid/src/core.rs

```diff
@@ -980,6 +1019,27 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                 // This event was meant only for our parent, so don't recurse.
                 false
             }
+            LifeCycle::DisabledChanged(ancestors_disabled) => {
+                let was_disabled = self.state.is_disabled();
+
+                self.state.is_explicitly_disabled = self.state.is_explicitly_disabled_new;
+                self.state.ancestor_disabled = *ancestors_disabled;
+
+                if was_disabled != self.state.is_disabled() {
+                    // In case we change but none of our children does we still need to update the
+                    // focus-chain
+                    self.state.reset_focus_chain();
+                    extra_event = Some(LifeCycle::DisabledChanged(self.state.is_disabled()));
+                } else if self.state.children_disabled_changed {
+                    self.state.reset_focus_chain();
+                    extra_event =
+                        Some(LifeCycle::Internal(InternalLifeCycle::RouteDisabledChanged));
```

Comment:
okay I think I also missed this in my first pass, and I'm having trouble wrapping my head around it.

When do we want to switch from `DisabledChanged` back to `RouteDisabledChanged`?

Reply:
I probably should write that in the code

Reply:
Do we actually need to route in this case? The child was disabled previously and remains disabled. Or is this about ensuring that the internal state is correctly reset?

Reply:
Oh, i forgot about that. It should be possible to leave the state dirty since it is already disabled. Even if is_explicitly_disabled is false and is_explicitly_disabled_new is true, this should get handled correctly and the widget would stay disabled when it receives `DisabledChanged(false)`. But it feels like moving the complexity to another place, because now it is possible to receive events while the state is dirty. And i can't tell how this would affect other things.